spec/components/all_components_spec.rb

```diff
@@ -0,0 +1,48 @@
+RSpec.describe "AllComponents", type: :controller do
+  Dir.glob("app/views/components/*.erb").each do |filename|
+    template = filename.split("/").last
+    component_name = template.sub("_", "").sub(".html", "").sub(".erb", "").gsub("-", "_")
```

Comment:
It's because the component names are loaded from their partials, so the sub removes the first _, then the gsub at the end converts the rest of the name from kebab case to snake case. I think for the moment we should keep this (because it matches the corresponding code in govuk_publishing_components), but we should look later at making all of these shared examples that are loaded directly from the components gem.